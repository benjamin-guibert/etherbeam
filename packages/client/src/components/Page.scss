@use '../styles/resources/sizes';

.my-page {
  margin: sizes.$space sizes.$space-l;
  position: relative;

  &-titleloader {
    position: absolute;
    top: sizes.$space;
    left: sizes.$space-s + sizes.$space;
  }

  &-subheader {
    display: flex;
  }
}
