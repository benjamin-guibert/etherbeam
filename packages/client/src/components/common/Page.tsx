import React, { ReactElement, ReactNode } from 'react'
import Container from 'react-bootstrap/Container'

interface PageProps {
  title: string
  children: ReactNode
}

const Page = ({ title, children }: PageProps): ReactElement => {
  return (
    <Container>
      <h1>{title}</h1>
      {children}
    </Container>
  )
}

export default Page
