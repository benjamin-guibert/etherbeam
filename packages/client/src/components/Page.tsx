import React, { ReactElement, ReactNode } from 'react'
import Loader from './Loader'
import './Page.scss'

type WidthLimitProp = 'xs' | 's' | 'm' | 'l'

interface PageProps {
  title: string | ReactNode
  widthLimit?: WidthLimitProp
  titleLoading?: boolean
  children?: ReactNode
}

const Page = ({ title, children, widthLimit = 's', titleLoading }: PageProps): ReactElement => {
  return (
    <main className={`my-page ${widthLimit ? `my-mw-${widthLimit}` : ''}`}>
      <h1 className="my-page-header">
        {titleLoading ? (
          <div className="my-page-titleloader">
            <Loader type="dots" />
          </div>
        ) : (
          title
        )}
      </h1>
      {children}
    </main>
  )
}

export default Page
