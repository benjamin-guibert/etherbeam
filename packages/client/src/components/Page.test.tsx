import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import Page from './Page'

describe('<Page />', () => {
  let component: ShallowWrapper

  describe('Default', () => {
    beforeAll(() => {
      component = shallow(<Page title="Title" />)
    })

    it('should render main', () => {
      const main = component.find('main')
      expect(main.hasClass('my-page')).toBeTruthy()
      expect(main.hasClass('my-mw-s')).toBeTruthy()
    })

    it('should render title', () => {
      const h1 = component.find('h1')
      expect(h1.hasClass('my-page-header')).toBeTruthy()
      expect(h1.find('div').exists()).toBeFalsy()
      expect(h1.prop('children')).toBe('Title')
      expect(h1.find('Loader').exists()).toBeFalsy()
    })

    it('should match snapshot', () => expect(component.render()).toMatchSnapshot())
  })

  describe('Title loading', () => {
    beforeAll(() => {
      component = shallow(<Page title="Title" titleLoading />)
    })

    it('should render title', () => {
      const h1 = component.find('h1')
      expect(h1.hasClass('my-page-header')).toBeTruthy()
      const div = h1.find('div').at(0)
      expect(div.hasClass('my-page-titleloader'))
      expect(div.find('Loader').exists()).toBeTruthy()
    })
  })
})
