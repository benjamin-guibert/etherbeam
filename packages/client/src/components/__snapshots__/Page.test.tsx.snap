// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page /> Default should match snapshot 1`] = `
<main
  class="my-page my-mw-s"
>
  <h1
    class="my-page-header"
  >
    Title
  </h1>
</main>
`;
